src/Microsoft.AspNet.TestHost/TestServer.cs

```diff
@@ -14,18 +14,20 @@
 
 namespace Microsoft.AspNet.TestHost
 {
-    public class TestServer : IServerFactory, IDisposable
+    public class TestServer : IDisposable
     {
         private const string DefaultEnvironmentName = "Development";
         private const string ServerName = nameof(TestServer);
-        private static readonly IFeatureCollection ServerInfo = new FeatureCollection();
-        private Func<IFeatureCollection, Task> _appDelegate;
+        private RequestDelegate _appDelegate;
         private IDisposable _appInstance;
         private bool _disposed = false;
+        private IHttpContextAccessor _httpContextAccessor;
```

Comment:
This looks wrong, you shouldn't have to do any of this. It's using the WebHostBuilder under the covers which does the right thing.